File: JavaScript/PascalTriangle.js
var generate = function(numRows) {

    let triangle = [];
    let firstRow = [1];

    if (numRows === 0)
        return triangle;

    triangle.push(firstRow);

    for (let i = 1; i < numRows; i++) {
        let row = [];
        let prevRow = triangle[i - 1];

        row.push(1);
        for (let j = 1; j < i; j++) {
            row.push(prevRow[j - 1] + prevRow[j]);
        }
        row.push(1);
        triangle.push(row);
    }

    return triangle;
};

File: C#/PascalTriangle.cs
public class PascalTriangle {

    public List<List<int>> generateTriangle(int numRows) {
        List<List<int>> triangle = new List<List<int>>();
        
        if (numRows == 0) {
            return triangle;
        }
        
        List<int> firstRow = new List<int>();
        
        firstRow.Add(1);
        triangle.Add(firstRow);
        
        for (int i = 1; i < numRows; i++) {
            
            List<int> row = new List<int>();
            List<int> prevrow = triangle[i - 1];
            
            row.Add(1);
            
            for (int j = 1; j < i; j++) {
                row.Add(prevrow[j - 1] + prevrow[j]);
            }
            
            row.Add(1);
            
            triangle.Add(row);
        }
        
        return triangle;
    }
}